app/controllers/charityInformation/CharityInformationSummaryController.scala

```diff
@@ -52,9 +54,31 @@ class CharityInformationSummaryController @Inject()(
   def onSubmit: Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
 
     for {
-      updatedAnswers <- Future.fromTry(result = request.userAnswers.set(Section1Page, true))
+      updatedAnswers <- Future.fromTry(result = checkComplete(request.userAnswers) match {
+        case None => Success(request.userAnswers)
+        case inProgressOrComplete => request.userAnswers.set(Section1Page, inProgressOrComplete.get)
```

Comment:
could you please use Some(inProgressOrComplete ) to remove ".get" reference at the end